src/mqt/bench/benchmark_generation.py

```diff
@@ -72,7 +67,7 @@ def generate_filename(
     benchmark_name: str,
     level: str,
     num_qubits: int | None,
-    provider_name: str | None = None,
+    gateset_name: str | None = None,
     device_name: str | None = None,
```

Comment:
I think this interface could be typed a little stronger.
Instead of strings for the gateset and the device name, I'd rather pass a `Gateset` and a `Device` object here.